model-sync-gradle/src/main/kotlin/org/modelix/model/sync/gradle/ModelSyncGradlePlugin.kt

```diff
@@ -0,0 +1,220 @@
+            it.inputDir.set(jsonDir)
+            it.registeredLanguages.set(syncDirection.registeredLanguages)
+            val serverTarget = syncDirection.target as ServerTarget
+
+            it.url.set(serverTarget.url)
+            it.repositoryId.set(serverTarget.repositoryId)
+            it.branchName.set(serverTarget.branchName)
+        }
+
+        project.tasks.register("runSync${syncDirection.name.replaceFirstChar { it.uppercaseChar() }}") {
+            it.dependsOn(importIntoModelServer)
+            it.group = "modelix"
+        }
+    }
+
+    private fun registerTasksForLocalTarget(
+        syncDirection: SyncDirection,
+        project: Project,
+        previousTask: TaskProvider<*>,
+        jsonDir: File,
+    ) {
+        val localTarget = syncDirection.target as LocalTarget
+
+        val antScript = jsonDir.resolve("build.xml")
+
+        val antDependencies = project.configurations.create("model-import-ant-dependencies")
+        project.dependencies.add(antDependencies.name, "org.apache.ant:ant-junit:1.10.12")
+
+        val generateAntScriptName = "${syncDirection.name}GenerateAntScriptForImport"
+        val generateAntScript = project.tasks.register(generateAntScriptName, GenerateAntScriptForMps::class.java) {
+            it.dependsOn(previousTask)
+            it.mpsHomePath.set(localTarget.mpsHome?.absolutePath)
+            it.mpsHeapSize.set(localTarget.mpsHeapSize)
+            it.repositoryPath.set(localTarget.repositoryDir?.absolutePath)
+            it.antScriptFile.set(antScript)
+            it.mpsDependenciesPath.set(getDependenciesDir(project).absolutePath)
+            it.jsonDirPath.set(jsonDir.absolutePath)
+            it.exportFlag.set(false)
+            it.includedModules.set(emptyList())
+        }
+
+        val importName = "${syncDirection.name}ImportIntoMps"
+        val importIntoMps = project.tasks.register(importName, ImportIntoMps::class.java) {
+            it.dependsOn(generateAntScript)
+            it.workingDir = jsonDir
+            it.classpath = project.getAntDependencies()
+            it.mainClass.set("org.apache.tools.ant.launch.Launcher")
+            it.jsonDir.set(jsonDir)
+            it.antScript.set(jsonDir.resolve("build.xml"))
+            it.mpsHome.set(localTarget.mpsHome)
+        }
+
+        project.tasks.register("runSync${syncDirection.name.replaceFirstChar { it.uppercaseChar() }}") {
+            it.dependsOn(importIntoMps)
+            it.group = "modelix"
+        }
+    }
+
+    private fun getBaseDir(project: Project): File {
+        return project.buildDir.resolve("model-sync")
+    }
+
+    private fun getDependenciesDir(project: Project): File {
+        return getBaseDir(project).resolve("dependencies")
+    }
+
+    private fun readModelixCoreVersion(): String? {
+        val resources: Enumeration<URL>? = javaClass.classLoader.getResources("modelix.core.version.properties")
+        while (resources != null && resources.hasMoreElements()) {
```

Comment:
```suggestion
        val resources: Enumeration<URL>? = javaClass.classLoader.getResources("modelix.core.version.properties") ?: return null
        if (resources.hasMoreElements()) {
```

There's no need for the loop. The later code only takes the first element if it exists.